pallets/circuit/src/square_up/mod.rs

```diff
@@ -0,0 +1,244 @@
+        for mut fsx in step_fsx.iter_mut() {
+            let sfx_id = fsx.generate_id::<SystemHashing<T>, T>(local_ctx.xtx_id);
+            if let Some(bid) = &fsx.best_bid {
+                if !<T as Config>::AccountManager::assign_deposit(sfx_id, &bid.executor) {
+                    log::error!(
+                        "assign_deposit: expect assign_deposit to succeed for sfx_id: {:?}",
+                        sfx_id
+                    );
+                } else {
+                    fsx.input.enforce_executor = Some(bid.executor.clone());
+                    res = true;
+                }
+            } else {
+                log::error!(
+                    "bind_bidders: expect best_bid to be Some for sfx_id: {:?}",
+                    sfx_id
+                );
+            }
+        }
+
+        res
+    }
+
+    /// Drop Xtx and unlock requester and all executors that posted bids - without penalties.
+    pub fn kill(local_ctx: &LocalXtxCtx<T>) -> bool {
+        let mut killed = false;
+        for fsx in Machine::<T>::read_current_step_fsx(local_ctx).iter() {
+            let sfx_id = fsx.generate_id::<SystemHashing<T>, T>(local_ctx.xtx_id);
+            if !<T as Config>::AccountManager::cancel_deposit(sfx_id) {
+                log::error!(
+                    "kill: expect cancel_deposit to succeed for sfx_id: {:?}",
+                    sfx_id
+                );
+            } else {
+                killed = true;
+            }
+            if let Some(bid) = &fsx.best_bid {
+                if !<T as Config>::AccountManager::cancel_deposit(
+                    bid.generate_id::<SystemHashing<T>, T>(sfx_id),
+                ) {
+                    log::error!(
+                        "kill: expect cancel_deposit to succeed for bid_id: {:?}",
+                        bid.generate_id::<SystemHashing<T>, T>(sfx_id)
+                    );
+                }
+            }
+        }
+        killed
+    }
+
+    /// Finalize Xtx after successful run.
+    pub fn finalize(local_ctx: &LocalXtxCtx<T>) -> bool {
+        let optimistic_fsx_in_step: Vec<
+            FullSideEffect<
+                <T as frame_system::Config>::AccountId,
+                <T as frame_system::Config>::BlockNumber,
+                BalanceOf<T>,
+            >,
+        > = Machine::<T>::read_current_step_fsx(local_ctx)
+            .iter()
+            .filter(|&fsx| fsx.security_lvl == SecurityLvl::Optimistic)
+            .cloned()
+            .collect();
+
+        let mut finalized = true;
+
+        let mut step_outcome = Outcome::Commit;
+
+        for fsx in optimistic_fsx_in_step.iter() {
+            let sfx_id = fsx.generate_id::<SystemHashing<T>, T>(local_ctx.xtx_id);
+            match &fsx.best_bid {
+                Some(bid) => {
+                    let outcome = match &fsx.confirmed {
+                        // Revert deposits for honest SFX resolution
+                        Some(_confirmed) => Outcome::Revert,
```

Comment:
Shouldn't this be ´Outcome::Commit´ here?

Reply:
For each honest execution each of executors have their bonded deposits (in the form of RequestCharge) returned (Outcome::Revert), whereas Requester proceeds with Settlements via Outcome::Commit. so this is correct